### ISSUE TITLE
Support `.mjs` configuration files

### ISSUE THREAD

**Issue by ybiquitous** [stylelint/stylelint#6905]
### What is the problem you're trying to solve?

Cosmiconfig has [supported `.mjs` files since v8.2](https://github.com/cosmiconfig/cosmiconfig/blob/v8.2.0/CHANGELOG.md). I believe it'd be useful if Stylelint also could support it.

### What solution would you like to see?

First, we may need to resolve a problem with loaders. It may not be easy.

https://github.com/stylelint/stylelint/blob/2413e8f186015a0a546ab1fa69c077a196141dd0/lib/createStylelint.js#L22-L27

Then, we must update the CLI help and documentation below:

https://github.com/stylelint/stylelint/blob/2413e8f186015a0a546ab1fa69c077a196141dd0/lib/cli.js#L115-L118

https://github.com/stylelint/stylelint/blob/2413e8f186015a0a546ab1fa69c077a196141dd0/docs/user-guide/configure.md?plain=1#L3-L8

See also:
- #6898
- #6902
- #5291

**Comment by romainmenke** [comment]
> First, we may need to resolve a problem with loaders. It may not be easy.

I think this should just work.
I did no change the loader `.mjs` files so that we wouldn't have to make further changes.

The only hurdle is that we can't write tests for this with `jest`.

**Comment by ybiquitous** [comment]
Unfortunately, my try for `.mjs` fails. Did I miss something?

`package.json`:

```json
{
  "devDependencies": {
    "stylelint": "^15.7.0"
  },
  "scripts": {
    "test": "stylelint -c config.mjs *.css",
    "test-cjs": "stylelint -c config.cjs *.css"
  }
}
```

`config.mjs`:

```js
export default {
  rules: {
    "block-no-empty": true,
  },
};
```

`config.cjs`:

```js
module.exports = {
  rules: {
    "block-no-empty": true,
  },
};
```

`test.css`:

```css
a{}
```

Run:

```console
$ npm i
...

$ npm test
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/masafumi.koba/tmp/foo/config.mjs not supported.
Instead change the require of /Users/masafumi.koba/tmp/foo/config.mjs to a dynamic import() which is available in all CommonJS modules.
...

$ npm run test-cjs
test.css
 1:2  ✖  Unexpected empty block  block-no-empty

1 problem (1 error, 0 warnings)
```

**Comment by alexander-akait** [comment]
We can write test for `jest`, but it is tricky, we can use `vm` and run ESM in `vm`, also we need to run with some flags (not for jest, for Node.js, because it is still experiment), as another workaround we can spawn a process with only loading (i.e. create a small file with our loading API) and test it

**Comment by ybiquitous** [comment]
> Was that with or without the `v8` cache package?

Good catch! Using the HEAD version of `stylelint` can resolve the error: 😅 👍🏼 

```diff
  "devDependencies": {
-   "stylelint": "^15.7.0"
+   "stylelint": "github:stylelint/stylelint"
  },
```

> We can write test for `jest`, but it is tricky

It might be ideal for writing tests, but I don't think writing tests for this feature is essential because writing tricky tests for the feature Cosmiconfig provides has very few benefits.

---

So, we can just update the CLI help and the document to close this issue!

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.
